hazelcast/src/main/java/com/hazelcast/config/ExternalDataStoreConfig.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.Properties;
+
+import static com.hazelcast.internal.util.Preconditions.checkHasText;
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+
+/**
+ * Contains configuration for external data store
+ */
+public class ExternalDataStoreConfig implements NamedConfig {
+
+    private String name;
+    private String className;
+    private boolean shared;
+    private Properties properties = new Properties();
+
+    public ExternalDataStoreConfig() {
+    }
+
+    @Override
+    public ExternalDataStoreConfig setName(String name) {
+        this.name = checkNotNull(name, "Name must not be null");
```

Comment:
I like you added human-friendly message, not just "name" as it is in many places :)